tests/models/ranking/test_catboost_reranker.py

```diff
@@ -0,0 +1,205 @@
+                [40, 12, 1, "2021-11-19"],
+            ],
+            columns=Columns.Interactions,
+        )
+        user_id_map = IdMap.from_values([10, 20, 30, 40, 50, 60, 70, 80])
+        item_id_map = IdMap.from_values([11, 12, 13, 14, 15, 16])
+        interactions = Interactions.from_raw(interactions_df, user_id_map, item_id_map)
+        return Dataset(user_id_map, item_id_map, interactions)
+
+    def test_recommend_happy_path(self, dataset: Dataset) -> None:
+        cangen_1 = PopularModel()
+        cangen_2 = ImplicitItemKNNWrapperModel(CosineRecommender())
+
+        scores_fillna_value = -100
+        ranks_fillna_value = 3
+
+        candidate_generators = [
+            CandidateGenerator(cangen_1, 2, True, True, scores_fillna_value, ranks_fillna_value),
+            CandidateGenerator(cangen_2, 2, True, True, scores_fillna_value, ranks_fillna_value),
+        ]
+        splitter = TimeRangeSplitter("1D", n_splits=1)
+        sampler = PerUserNegativeSampler(1, 32)
+        two_stage_model_ranker = CandidateRankingModel(
+            candidate_generators,
+            splitter,
+            sampler=sampler,
+            reranker=CatBoostReranker(CatBoostRanker(random_state=32, verbose=False)),
+        )
+        two_stage_model_ranker.fit(dataset)
+
+        actual_reco_ranker = two_stage_model_ranker.recommend(
+            [10, 20, 30], dataset, k=3, filter_viewed=True, force_fit_candidate_generators=True
+        )
+        expected_reco_ranker = pd.DataFrame(
+            {
+                Columns.User: [10, 10, 20, 20, 20, 30],
+                Columns.Item: [14, 15, 12, 15, 13, 13],
+                Columns.Score: [
+                    11.909,
+                    1.020,
+                    23.396,
+                    1.020,
+                    -23.396,
+                    11.909,
+                ],
+                Columns.Rank: [1, 2, 1, 2, 3, 1],
+            }
+        )
+
+        two_stage_model_classifier = CandidateRankingModel(
+            candidate_generators,
+            splitter,
+            sampler=sampler,
+            reranker=CatBoostReranker(CatBoostClassifier(random_state=32, verbose=False)),
+        )
+        two_stage_model_classifier.fit(dataset)
+
+        actual_reco_classifier = two_stage_model_classifier.recommend(
+            [10, 20, 30], dataset, k=3, filter_viewed=True, force_fit_candidate_generators=True
+        )
+        expected_reco_classifier = pd.DataFrame(
+            {
+                Columns.User: [10, 10, 20, 20, 20, 30],
+                Columns.Item: [14, 15, 12, 15, 13, 13],
+                Columns.Score: [0.588, 0.505, 0.665, 0.505, 0.334, 0.588],
+                Columns.Rank: [1, 2, 1, 2, 3, 1],
+            }
+        )
+
+        pd.testing.assert_frame_equal(actual_reco_ranker, expected_reco_ranker, atol=0.001)
+        pd.testing.assert_frame_equal(actual_reco_classifier, expected_reco_classifier, atol=0.001)
```

Comment:
Maybe add a test to that different model instances on the same data give the same recommendations?

Reply:
This is exactly what this test is doing - we are checking that we receive the same result each time.
I don't think it's needed